src/Microsoft.MobileBlazorBindings.Authentication/Services/Apple/AppleAuthenticationService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.AspNetCore.Components.Authorization;
+using Microsoft.Extensions.Options;
+using Microsoft.MobileBlazorBindings.ProtectedStorage;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Essentials;
+
+namespace Microsoft.MobileBlazorBindings.Authentication
+{
+    /// <summary>
+    /// The default implementation for <see cref="IAuthenticationService"/> that uses IdentityModel.OidcClient to authenticate the user.
+    /// </summary>
+    /// <typeparam name="TRemoteAuthenticationState">The state to preserve across authentication operations.</typeparam>
+    /// <typeparam name="TAccount">The type of the <see cref="RemoteUserAccount" />.</typeparam>
+    /// <typeparam name="TProviderOptions">The options to be passed down to the underlying Authentication library handling the authentication operations.</typeparam>
+    public class AppleAuthenticationService<TRemoteAuthenticationState, TAccount, TProviderOptions> :
+        OidcAuthenticationService<TRemoteAuthenticationState, TAccount, TProviderOptions>
+        where TRemoteAuthenticationState : OidcAuthenticationState, new()
+        where TProviderOptions : new()
+        where TAccount : RemoteUserAccount
+    {
+        /// <summary>
+        /// Initializes a new instance.
+        /// </summary>
+        /// <param name="options">The options to be passed down to the underlying Authentication library handling the authentication operations.</param>
+        /// <param name="tokenCache">The token cache to use to store tokens.</param>
+        /// <param name="protectedStorage">The protect storage where refresh tokens will be stored.</param>
+        /// <param name="accountClaimsPrincipalFactory">The <see cref="AccountClaimsPrincipalFactory{TAccount}"/> used to generate the <see cref="ClaimsPrincipal"/> for the user.</param>
+        public AppleAuthenticationService(
+            IOptionsSnapshot<RemoteAuthenticationOptions<TProviderOptions>> options,
+            ITokenCache tokenCache,
+            IProtectedStorage protectedStorage,
+            AccountClaimsPrincipalFactory<TAccount> accountClaimsPrincipalFactory) : base(options, tokenCache, protectedStorage, accountClaimsPrincipalFactory)
+        {
+        }
+
+        protected override async Task<string> StartSecureNavigation(Uri startUrl, Uri redirectUrl)
```

Comment:
Is that any different from AndroidAuthenticationService?

Reply:
Not anymore. It used to be. But this might be different in the future again because of the possibility to toggle inprivate browsing on iOS. I'm preparing a PR for that for Xamarin, but if they don't take it, I'll write my own again. I left it in here for now. We'll see where it ends up ;-)